src/HealthChecks.MongoDb/DependencyInjection/MongoDbHealthCheckBuilderExtensions.cs

```diff
@@ -190,4 +190,124 @@ public static class MongoDbHealthCheckBuilderExtensions
             tags,
             timeout));
     }
+
+    /// <summary>
+    /// Add a health check for MongoDb that list all databases from specified <paramref name="mongoClientSettingsFactory"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="mongoClientSettingsFactory">A factory to build MongoClientSettings to be used.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'mongodb' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddMongoDb(
+        this IHealthChecksBuilder builder,
+        Func<IServiceProvider, MongoClientSettings> mongoClientSettingsFactory,

```

Comment:
How common is this scenario? (Having the MongoClientSettings come from an IServiceProvider). Is it something we could live without these overloads? Someone could easily do this themselves since MongoDbHealthCheck is public.

Reply:
I can see some use cases for that, the idea here was to give more flexibility and different options to configure the healthcheck.
But I think we can live without that and I've already removed the overloads.
Please review.

Reply:
Hi @eerhardt can we close this one?

Reply:
Yep, resolved.